src/self-managed-debug/src/system_catalog_dumper.rs

```diff
@@ -0,0 +1,167 @@
+        for service in services.list(&Default::default()).await? {
+            if let Some(spec) = service.spec {
+                if service.metadata.name.is_none() {
+                    continue;
+                }
+                let service_name = service.metadata.name.expect("checked above");
+
+                for port_info in spec.ports.unwrap_or_default() {
+                    let port = port_info.port;
+                    // Look for port with the target port specified in the CLI
+                    if args.sql_target_port.is_some()
+                        && args.sql_target_port.expect("checked above") == port
+                    {
+                        return Ok(SqlPortForwardingInfo {
+                            namespace: namespace.clone(),
+                            service_name,
+                            target_port: port,
+                            local_port: args.sql_local_port.unwrap_or(port),
+                        });
+                    }
+
+                    // Look for port named "internal-sql" or similar
+                    // By default, we use the internal SQL port.
+                    if let Some(port_name) = port_info.name {
+                        if port_name.to_lowercase().contains("internal")
+                            && port_name.to_lowercase().contains("sql")
+                        {
+                            return Ok(SqlPortForwardingInfo {
+                                namespace: namespace.clone(),
+                                service_name,
+                                target_port: port,
+                                local_port: args.sql_local_port.unwrap_or(port),
+                            });
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    Err(anyhow::anyhow!("No SQL port forwarding info found"))
+}
+
+/// Spawns a port forwarding process for the given k8s service.
+/// The process will retry if the port-forwarding fails and
+/// will terminate once the port forwarding reaches the max number of retries.
+/// We retry since kubectl port-forward is flaky.
+#[must_use]
+pub fn spawn_sql_port_forwarding_process(
+    port_forwarding_info: &SqlPortForwardingInfo,
+    k8s_context: Option<String>,
+) -> AbortOnDropHandle<()> {
+    let port_forwarding_info = port_forwarding_info.clone();
+
+    task::spawn(|| "port-forwarding", async move {
+        if let Err(err) = retry::Retry::default()
+            .max_duration(Duration::from_secs(60))
+            .retry_async(|retry_state| {
+                let k8s_context = k8s_context.clone();
+                let namespace = port_forwarding_info.namespace.clone();
+                let service_name = port_forwarding_info.service_name.clone();
+                let local_port = port_forwarding_info.local_port;
+                let target_port = port_forwarding_info.target_port;
+
+                println!(
```

Comment:
Instead of `println!(...)` should we be using `tracing::info!`? Maybe a nice follow up?